apps/db/lib/db/resource_unavailability.ex

```diff
@@ -25,4 +25,76 @@ defmodule DB.ResourceUnavailability do
     |> limit(1)
     |> Repo.one()
   end
+
+  def availability_over_last_days(%Resource{} = resource, nb_days) when is_integer(nb_days) and nb_days > 0 do
+    %{hours: hours} = unavailabilities_over_last_days(resource, nb_days)
+    round_float(100 - hours / (24.0 * nb_days) * 100)
+  end
+
+  @doc """
+  Round a float up to a precision and removes unneeded zeroes.
+
+  iex> round_float(1.23)
+  1.2
+
+  iex> round_float(1.0)
+  1
+
+  iex> round_float(1.20, 2)
+  1.20
+  """
+  def round_float(float, precision \\ 1) do
+    rounded = Float.round(float, precision)
+    trunced = trunc(float)
+    if rounded == trunced, do: trunced, else: rounded
+  end
+
+  def unavailabilities_over_last_days(%Resource{id: resource_id}, nb_days) when is_integer(nb_days) and nb_days > 0 do
+    # Sorry about the messy Ecto code
+    # - interval to number of hours, https://stackoverflow.com/a/952600
+    rows =
+      __MODULE__
+      |> select([:start, :end])
+      |> where([r], r.resource_id == ^resource_id)
+      |> where(
+        [r],
+        fragment(
+          ~s[now() between "start" - '1 day'::interval * ? and greatest(coalesce("end", now()), now())],
```

Comment:
Si vous connaissez une manière plus élégante

Reply:
Cette condition est problématique car toujours vraie. En effet `"start" - '1 day'::interval * ?` est une date passée. `greatest(coalesce("end", now()), now())` vaut toujours now() car "end" est une date passée. Donc now() est toujours entre les 2.

Je pense qu'il faut mettre `~s["end" IS NULL OR "end" between now() - '1 day'::interval * ? and now()]`

Reply:
j'ai rajouté un test qui montre le problème, quand on a une période d'indisponibilité passée (avant nb_days), on se retrouve avec un pourcentage de dispo > 100%

Reply:
Merci beaucoup d'avoir mis le doigt dessus et d'avoir ajouté un test !